MAUI/DataForm/layout.md

```diff
@@ -0,0 +1,750 @@
+
+    private void OnGenerateDataFormItem(object sender, GenerateDataFormItemEventArgs e)
+    {
+        if (e.DataFormItem != null)
+        {
+            if (e.DataFormItem.FieldName == "Name")
+            {
+                e.DataFormItem.LeadingLabelIcon = ImageSource.FromFile("ContactInfo.png");
+            }
+        }
+    }
+
+{% endhighlight %}
+{% endtabs %}
+
+## Changing order of the DataFormItem
+
+The order of the `DataFormItem` can be changed by using attributes or by handling `GenerateDataFormItem` event.
+
+### Using attributes
+
+The fields order can be changed by using the `RowOrder` property in [DataFormDisplayOptionsAttribute] attribute.
+
+{% tabs %}
+{% highlight c# %}
+
+    [DataFormDisplayOptions(RowOrder = 1)]
+    public string Adress { get; set; }
+
+    [DataFormDisplayOptions(RowOrder = 0)]
+    public string Name { get; set; }
+
+{% endhighlight %}
+{% endtabs %}
+
+### Using event
+
+The fields order can be changed by using the [RowOrder] property in the `DataFormItem`.
+
+{% tabs %}
+{% highlight c# %}
+
+this.DataForm.GenerateDataFormItem += OnGenerateDataFormItem;
+
+    private void OnGenerateDataFormItem(object sender, GenerateDataFormItemEventArgs e)
+    {
+
+        if (e.DataFormItem != null)
+        {
+            if (e.DataFormItem.FieldName == "Name")
+            {
+                e.DataFormItem.RowOrder = 0;
+            }
+        }
+    }
+
+{% endhighlight %}
+{% endtabs %}
+
+## Changing order of the DataFormItem in same row
+
+The order of the `DataFormItem` can be changed within a row by using attributes or by handling `GenerateDataFormItem` event.
+
+### Using attributes
+
+The order of the fields can be changed by using the `ItemsOrderInRow` property in [DataFormDisplayOptionsAttribute] attribute. This is applicable only for Grid layout.
+
+{% tabs %}
+{% highlight c# %}
+
+    [DataFormDisplayOptions(RowOrder = 0, ItemsOrderInRow = 1)]
+    public string LastName { get; set; }
+
+    [DataFormDisplayOptions(RowOrder = 0, ItemsOrderInRow = 0)]
+    public string FirstName { get; set; }
+
+{% endhighlight %}
+{% endtabs %}
+
+### Using event
+
+The order of the fields can be changed by using the [ItemsOrderInRow] property in the `DataFormItem`.
+
+{% tabs %}
+{% highlight c# %}
+
+this.DataForm.GenerateDataFormItem += OnGenerateDataFormItem;
+
+    private void OnGenerateDataFormItem(object sender, GenerateDataFormItemEventArgs e)
+    {
+
+        if (e.DataFormItem != null)
+        {
+            if (e.DataFormItem.FieldName == "FirstName")
+            {
+                e.DataFormItem.ItemsOrderInRow = 1;
+            }
+            if (e.DataFormItem.FieldName == "LastName")
+            {
+                e.DataFormItem.ItemsOrderInRow = 0;
+            }
+        }
+    }
+
+{% endhighlight %}
+{% endtabs %}
+
+## Grouping data fields
+
+It is possible to group some fields and set group name in the data form. You can expand or collapse the group by tapping the group item.
```

Comment:
The .NET MAUI DataForm support to group the editors which are relevant to each other. You can expand or collapse the group by tapping the group item.